test/internal-events/fixtures/ava.config.js

```diff
@@ -0,0 +1,16 @@
+import fs from 'node:fs/promises';
+
+const internalEvents = [];
+
+export default {
+	files: [
+		'test.js',
+	],
+	async onInternalEvent(event) {
```

Comment:
How about passing an async generator?

```suggestion
	async observeRun(runs) {
	  for await (const event of run.events) {
```

Reply:
yeah that makes sense